### PR TITLE
[Regression] Re-enable ANSI color support on Windows

### PR DESCRIPTION
**Pull request by goderbauer**
Also adds a test to make sure we don't break it again.

### CODE REVIEW

File: dev/devicelab/pubspec.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ dependencies:
   args: ^0.13.4
   meta: ^1.0.4
   path: ^1.4.0
-  process: 2.0.1
+  process: 2.0.3
```
Comment (Hixie) [inline dev/devicelab/pubspec.yaml]: is this related to the rest of the patch?

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 2]
```diff
@@ -275,7 +266,7 @@ class AnsiTerminal {
     _INVALID_HANDLE,
   ];
 
-  bool supportsColor;
+  bool supportsColor = platform.stdoutSupportsAnsi;
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: where is this defined?

Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: never mind, found it

File: dev/devicelab/pubspec.yaml (hunk 1 above)
Comment (goderbauer) [inline dev/devicelab/pubspec.yaml]: Yes. process depends on platform. When I bumped platform I also had to update process. Otherwise pub's dependency solver is unhappy.

File: packages/flutter_tools/lib/src/base/logger.dart [hunk 3]
```diff
@@ -248,15 +248,6 @@ enum _LogType {
 }
 
 class AnsiTerminal {
-  AnsiTerminal() {
-    if (platform.isWindows) {
-      supportsColor = platform.ansiSupported;
-    } else {
-      final String term = platform.environment['TERM'];
-      supportsColor = (term != null && term != 'dumb');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: removing this will break flutter in emacs

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 2 above)
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/logger.dart]: In package:platform: https://github.com/google/platform.dart/blob/master/lib/src/interface/local_platform.dart#L54

File: packages/flutter_tools/test/src/base/logger_test.dart [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  group('AnsiLogger', () {
+    test('should support color on Windows version >= 10.0.10586', () {
+      final String name = new OperatingSystemUtils().name;
+      final String versionString = new RegExp(r'\d+\.\d+\.\d+').firstMatch(name).group(0);
+      final Version version = new Version.parse(versionString);
+      final bool shouldSupportColor = version >= new Version.parse('10.0.10586');
+
+      expect(new AnsiTerminal().supportsColor, shouldSupportColor);
+    }, testOn: 'windows');
+  });
```
Comment (Hixie) [inline packages/flutter_tools/test/src/base/logger_test.dart]: can you add a second test that verifies that if TERM=dumb, it doesn't support color? :-)

Comment (Hixie) [inline packages/flutter_tools/test/src/base/logger_test.dart]: or maybe just a TODO to add the test, since that test is a pain to write

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/base/logger.dart]: @goderbauer showed me that dart:io claims it won't. we shall see!

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/base/logger.dart (hunk 3 above)
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/logger.dart]: According to the Dart Docs it shouldn't: https://api.dartlang.org/dev/1.23.0-dev.11.5/dart-io/Stdout/supportsAnsiEscapes.html

File: packages/flutter_tools/test/src/base/logger_test.dart (hunk 4 above)
Comment (goderbauer) [inline packages/flutter_tools/test/src/base/logger_test.dart]: That's not easy to do: the environment is fixed when the dart VM process is started, so I have no chance of injecting TERM=dumb into the env early enough for the current dart VM to pick it up. I'd have to call out to another process, which seems a bit to heavy for this unit test.
